File: src/crne/Crne.java
package crne;

import java.util.Scanner;

// https://open.kattis.com/problems/crne

public class Crne {

    public static void main(String[] args) {

        String in = "3";
        Scanner scan = new Scanner(in);

        long number = scan.nextLong();
        long result = maximizeShatter(number);

        System.out.println(result);
    }

    static long maximizeShatter(long in) {
        long cutHorizontal = in / 2;
        long cutVertical   = in - cutHorizontal;
        return (cutHorizontal + 1) * (cutVertical + 1);
    }
}

File: src/crne/crne.go
package main

import (
	"fmt"
)

// https://open.kattis.com/problems/crne

func main() {
	var cuts int
	fmt.Scanln(&cuts)

	cutHoriz := cuts / 2
	cutVert := cuts - cutHoriz

	fmt.Println((cutHoriz+1) * (cutVert+1))
}
